File: windows/Scanner/Rules/AESCFG002.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESCFG002 : IRule
{
    public string Id => "AES-CFG-002";
    public Severity Severity => Severity.Medium;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts) =>
        facts.McpServers
            .Where(server => !server.Disabled)
            .GroupBy(server => server.Name, StringComparer.OrdinalIgnoreCase)
            .Where(group => group.Select(server => server.AppId).Distinct(StringComparer.Ordinal).Count() > 1)
            .SelectMany(group => group.Select(server => this.Finding(server.AppId, server.Name, affectedPath: server.ConfigPath)))
            .ToArray();
}

File: macos/Scanner/Sources/Scanner/Rules/AESCFG002.swift
public struct AESCFG002: Rule {
    public let id = "AES-CFG-002"
    public let severity: Severity = .medium

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        let activeServers = facts.mcpServers.filter { !$0.disabled }
        let grouped = Dictionary(grouping: activeServers, by: { $0.name.lowercased() })
        let duplicates = grouped.values.filter { servers in
            Set(servers.map(\.appId)).count > 1
        }

        return duplicates.flatMap { servers in
            servers.map { server in
                finding(app: server.appId, serverName: server.name, affectedPath: server.configPath)
            }
        }
    }
}
